partner_delivery_zone/models/sale_order.py

```diff
@@ -19,5 +19,4 @@ class SaleOrder(models.Model):
     @api.depends("partner_shipping_id")
     def _compute_delivery_zone_id(self):
         for so in self:
-            if so.partner_shipping_id.delivery_zone_id:
-                so.delivery_zone_id = so.partner_shipping_id.delivery_zone_id
+            so.delivery_zone_id = so.partner_shipping_id.delivery_zone_id
```

Comment:
That `if` was done on purpose if I don't remember bad, so you may need:
```suggestion
            if so.partner_shipping_id.delivery_zone_id:
                so.delivery_zone_id = so.partner_shipping_id.delivery_zone_id
            else:
                so.delivery_zone_id = so.delivery_zone_id  # reassign old value
```
(not sure this works)